src/irmin-pack-tools/ppidx/ppidx.ml

```diff
@@ -1,19 +1,34 @@
 module Hash = Irmin.Hash.SHA256
 module Index = Irmin_pack_unix.Index.Make (Hash)
 
-type t = { hash : string; off : Optint.Int63.t; len : int; kind : string }
+type hash = { base64 : string; base58 : string }
 [@@deriving irmin]
 
-let dump_json k v =
-  let hash = Base64.encode_exn (Index.Key.encode k) in
+type t = { hash : hash; off : Optint.Int63.t; len : int; kind : string }
+[@@deriving irmin]
+
+type alphabet = Bitcoin | Ripple | Flickr
+
+let make_alphabet p c =
+  match p, c with
+  | _, Some s -> B58.make_alphabet s
+  | Bitcoin, _ -> B58.make_alphabet "123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz"
+  | Ripple, _ -> B58.make_alphabet "rpshnaf39wBUDNEGHJKLM4PQRST7VWXYZ2bcdeCg65jkm8oFqi1tuvAxyz"
+  | Flickr, _ -> B58.make_alphabet "123456789abcdefghijkmnopqrstuvwxyzABCDEFGHJKLMNPQRSTUVWXYZ"
```

Comment:
Unfortunately, Tezos has a slightly different encoding; see https://github.com/mirage/irmin/blob/main/src/irmin-tezos/schema.ml#L32 and https://github.com/tarides/tezos-base58/blob/main/tezos_base58.ml#L123 (they use a prefix + they encode a checksum at the end of the hash). I would suggest using `tezos-base58` directly.

Reply:
Yeah I agree... Using `tezos-base58` would also sidestep the [two published `base58` packages](https://github.com/ocaml/opam-repository/issues/24160) >_>

Reply:
Is it possible to use the `pp` for the hash instead of doing this encoding ourselves?

Reply:
Ha yes indeed, using `Irmin.Type.pp Hash.t` is probably the best way to go!

Reply:
I changed the code, taking your comments into account:
It will now ask for a type of store, and infer the encoding of the hash from it (Irmin = SHA256, Tezos = BLAKE2B)
And then use `Irmin.Type.to_string`

Reply:
(Thank you art-w for your live comments on this)